dask/dataframe/io/parquet/arrow.py

```diff
@@ -546,9 +546,10 @@ def read_partition(
             use_pandas_metadata=True,
             use_threads=False,
             **kwargs.get("read", {}),
-        ).to_pandas(categories=categories, use_threads=False, ignore_metadata=True)[
-            list(columns_and_parts)
-        ]
+        ).to_pandas(categories=categories, use_threads=False, ignore_metadata=False)
+        if not isinstance(df.index, pd.RangeIndex):
+            df.reset_index(drop=False, inplace=True)
```

Comment:
We should maybe double-check what pandas' behaviour is in this case.
We *certainly* don't want to reset the index and then set it again a few lines down for a range-index. We also don't want to materialise it in the case it's a "default" (1..N) index that always got written and wasn't ever really data; but is there any case in which a range index might be useful?

Reply:
Can you clarify the concern? RangeIndex objects are cheap to create / assign.

And what happens if we just don't do the `reset_index` at all?

Reply:
So, the first bad case is materialising the column when either it isn't wanted at all, or that it ought to be the index after all, so should have stayed a range.

The second bad case is that the user wants a different column to be the index but **does** want the information in the range-index as a column, as useful data.

Reply:
Ah, I missed that we have `drop=False`. That explains why `reset_index` might be materializing data. Why would a column be written as the index in the pandas metadata, but not used as the index when reading?

I don't understand the second case. If the column should be returned it'll be in `columns_and_parts`, right?

Reply:
> Why would a column be written as the index in the pandas metadata, but not used as the index when reading?

I agree that this is unlikely to happen much (on purpose), but the functionality is still supported.  I suspect that this will only happen, in practice, if the user has a DataFrame with a default/unused index, and then writes a parquet dataset without using `write_index=False`.  In this case, they **may** end up specifying a different column as the index during the read.

the "second" bad case is just the scenario above, but the index column listed in the pandas metadata happens to be a column the user still wants in the data.  In this case, we need to reset the index so we can preserve that column.